### PR TITLE
[FEATURE] [MER-3703] Add cover image editing at the section level

### PR DESCRIPTION
**Pull request by rgachuhi**
This PR supports changing the section cover image in the course section's delivery interface. 
Screenshot 2024-09-23 at 1 41 06 PM

### CODE REVIEW

**Review by Francisco-Castro — CHANGES_REQUESTED** [review]
I tested the code locally, and it's working as expected. The only part I noticed is the flash message not appearing when successfully uploading an image.

Could you please take a look at it?

Note: I left a few suggestions for you."

File: lib/oli_web/live/sections/overview_view.ex [hunk 1]
```diff
@@ -418,6 +429,22 @@ defmodule OliWeb.Sections.OverviewView do
...
+          <ImageUpload.render
+            section={@section}
+            uploads={@uploads}
+            changeset={to_form(@changeset)}
+            upload_event="update_image"
+            change="change"
+            cancel_upload="cancel_upload"
+          />
+        </section>
+      </Group.render>
```
[the remark this answers is not in the document]
Comment (rgachuhi) [reply lib/oli_web/live/sections/overview_view.ex]: Suggested tweaks applied

File: lib/oli_web/live/sections/details/image_upload.ex [hunk 2]
```diff
@@ -0,0 +1,153 @@
+defmodule OliWeb.Sections.Details.ImageUpload do
```
[the remark this answers is not in the document]
Comment (rgachuhi) [reply lib/oli_web/live/sections/details/image_upload.ex:1]: thanks

File: lib/oli_web/live/sections/overview_view.ex [hunk 3]
```diff
@@ -628,6 +655,45 @@ defmodule OliWeb.Sections.OverviewView do
...
+        temp_file_path = meta.path
+        section_path = "sections/#{socket.assigns.section.slug}"
+        image_file_name = "#{entry.uuid}.#{ext(entry)}"
+        upload_path = "#{section_path}/#{image_file_name}"
+
+        {:ok, uploaded_file} = S3Storage.upload_file(bucket_name, upload_path, temp_file_path)
+        {:ok, uploaded_file}
+      end)
+
+    with uploaded_path <- Enum.at(uploaded_files, 0),
```
[the remark this answers is not in the document]
Comment (rgachuhi) [reply lib/oli_web/live/sections/overview_view.ex]: Suggestion applied

**Pull request merged**